src/sagas/background-refresh.js

```diff
@@ -174,12 +175,37 @@ function* refreshConsolePage ({ id }) {
   }
 }
 
-function* startSchedulerWithFixedDelay (action) {
+function* startSchedulerWithFixedDelay ({ payload: { delayInSeconds, startDelayInSeconds, ...rest } }) {
   // if a scheduler is already running, stop it
   yield put(Actions.stopSchedulerFixedDelay())
 
+  const lastRefresh = yield select(state => state.config.get('lastRefresh', 0))
+
   // run a new scheduler
-  yield schedulerWithFixedDelay(action.payload)
+  yield schedulerWithFixedDelay({
+    delayInSeconds,
+    startDelayInSeconds: calculateStartDelayIfMissing({ delayInSeconds, startDelayInSeconds, lastRefresh }),
+    ...rest,
+  })
+}
+
+/* Continue previous wait period (unless immediate refresh is forced).
+  Restarting the wait period could lead to irregular, long intervals without refresh
+  or prevent the refresh (as long as user will keep changing the interval)
+  Example:
+  1. previous refresh period is 2 min (1m 30sec already elapsed)
+  2. user changes it to 5min
+  3. already elapsed time will be taken into consideration and refresh will be
+     triggered after 3 m 30sec.
+  Result: Wait intervals will be 2min -> 2min -> 5min -> 5min.
+  With restarting timers: 2min -> 2min -> 6min 30 sec -> 5min.
+*/
+function calculateStartDelayIfMissing ({ delayInSeconds, startDelayInSeconds, lastRefresh }) {
```

Comment:
I don't quite see why this is really necessary.

Why wouldn't it be good enough to just reset the scheduler to run delayInSeconds in the future?

Reply:
IMHO it's worth it. It produces more deterministic intervals and fixes some border cases.